chore: upgrade from jquery-cropper to cropperjs v2

rajanarahul93 opened the pull request:
<!-- IMPORTANT CHECKLIST
Make sure you've done all the following (You can delete the checklist before submitting)
- [x] PR title is prefixed by one of the following: feat, fix, docs, style, refactor, test, build, ci, chore, revert, l10n, taxonomy
- [x] Code is well documented
- [x] Include unit tests for new functionality (not applicable here)
- [x] Code passes GitHub workflow checks in your branch
- [x] If you have multiple commits please combine them into one commit by squashing them.
- [x] Read and understood the [contribution guidelines](https://github.com/openfoodfacts/openfoodfacts-server/blob/main/CONTRIBUTING.md)
-->

### What

This PR removes the deprecated `jquery-cropper` dependency and migrates the image cropping functionality to `cropperjs` v2.

Changes made:
- Removed `jquery-cropper.js` reference from `gulpfile.ts`
- Refactored all `$(...).cropper()` jQuery-based usages to modern `Cropper` instance methods from `cropperjs` v2
- Updated JS file (`product-multilingual.js`) to use native DOM manipulation instead of jQuery

### Related issue

- Fixes #11874

hangy commented on html/js/product-multilingual.js:
```diff
@@ -211,10 +215,10 @@ function rotate_image(event) {
     angles[imagefield] = (360 + angles[imagefield]) % 360;
 
-    $('img#crop_' + imagefield).cropper('rotate', angle);
+    croppers[imagefield]?.rotate(angle);
```
It could be better to store the current cropper in a variable and exit early if it's `null`/`undefined`. Would make following calls easier by not having to use the null-forgiving operator.
Also, some later calls like `croppers[imagefield]?.getCropBoxData();` would not return anything usable, but aren't guarded against the cropper not existing for some reason

hangy commented on html/js/product-multilingual.js:
```diff
@@ -375,34 +379,50 @@ function change_image(imagefield, imgid) {
-        "zoomOnTouch": false,
-        "toggleDragModeOnDblclick": true,
-        "checkCrossOrigin": false
+    document
+      .getElementById("crop_" + imagefield)
+      .addEventListener("click", () => {
+        croppers[imagefield]?.clear();
+      });
+
+    const imageElement = document.getElementById("crop_" + imagefield);
```
The `imageElement` variable could be reused, for example:
```suggestion
    const imageElement = document.getElementById("crop_" + imagefield);
    imageElement.addEventListener("click", () => {
        croppers[imagefield]?.clear();
    });

```

hangy reviewed (commented):
Thanks for the PR! Just some minor suggestions.

I think you also need to remove this: https://github.com/openfoodfacts/openfoodfacts-server/blob/892662f5374432cb6f2a45cd4d0dd24b62955949/cgi/product_multilingual.pl#L818

rajanarahul93 commented:
Thank you for the feedback, @hangy 
I’ll go ahead and apply your suggestions, including removing the deprecated <script> tag from product_multilingual.pl.

hangy commented:
/update_tests_results

hangy commented on html/js/product-multilingual.js:
```diff
@@ -22,6 +22,8 @@
 /*eslint no-console: "off"*/
 /*global lang admin otherNutriments Tagify*/
 /*exported add_line upload_image update_image update_nutrition_image_copy*/
+import Cropper from "cropperjs";
```
> Uncaught SyntaxError: import declarations may only appear at top level of a module [product-multilingual.js:1:1](http://static.openfoodfacts.localhost/js/dist/product-multilingual.js?v=1748440389)

rajanarahul93 commented:
Hi @hangy ,

I've made the requested changes:
- Moved the import Cropper from "cropperjs" statement to the top of product-multilingual.js.
- Updated the corresponding script tag in product_multilingual.pl to use type="module".
- Added a console.warn message when a cropper instance is not found, as suggested.

Please let me know if there are any further changes required — I’d be happy to work on them!

Thanks!!

hangy reviewed (commented):
Thanks @rajanarahul93!

I had a look at your updated suggestion, and tried to test it. Apparently, the handler `$('img#crop_' + imagefield).on('ready', function () {` is never called, so that the buttons stay disabled. You probably want to take a look at the [new Cropper v2 API](https://fengyuanchen.github.io/cropperjs/api/) to see what else needs to be changed.

If you need help testing yourself, it's actually rather easy [with Docker](https://openfoodfacts.github.io/openfoodfacts-server/dev/how-to-quick-start-guide/ "How to setup the Dev environment (quick start guide)")

rajanarahul93 commented:
Thank you so much for the detailed feedback, @hangy

Sorry for the confusion — I’ll make the suggested changes right away:

Update the import path to use ./cropper.esm.js
Bundle cropper.esm.js via gulpfile.ts
Replace the rotate() call with the correct method from the new Cropper v2 API

I really appreciate your guidance! Let me know if there’s anything else — happy to work on it.

hangy commented on cgi/product_multilingual.pl:
```diff
@@ -815,7 +815,6 @@ ($product_ref, $field, $language, $request_ref)
 	$request_ref->{scripts} .= <<HTML
 <script type="text/javascript" src="$static_subdomain/js/dist/webcomponentsjs/webcomponents-loader.js"></script>
 <script type="text/javascript" src="$static_subdomain/js/dist/cropper.js"></script>
```
```suggestion
```

hangy commented on html/js/product-multilingual.js:
```diff
@@ -363,7 +378,8 @@ function change_image(imagefield, imgid) {
     });
 
-    $('img#crop_' + imagefield).on('ready', function () {
+        const cropperImage = cropper.getCropperImage();
```
`cropper` does not exist in this scope (yet). You later in line 400.

hangy commented on html/js/product-multilingual.js:
```diff
@@ -363,7 +378,8 @@ function change_image(imagefield, imgid) {
 
-    $('img#crop_' + imagefield).on('ready', function () {
+        const cropperImage = cropper.getCropperImage();
+        cropperImage.addEventListener("cropper:ready", function () {
```
How does this work? I can't find any docs regarding the `cropper:ready` event
